mcgill-augmented-reality-backend/src/main/java/mcgillar/backend/controllers/user/AppUserInfoController.java

```diff
@@ -0,0 +1,108 @@
+package mcgillar.backend.controllers.user;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import lombok.AllArgsConstructor;
+import mcgillar.backend.TO.user.UserInfoTO;
+import mcgillar.backend.model.user.AppUser;
+import mcgillar.backend.model.user.AppUserInfo;
+import mcgillar.backend.services.user.AppUserInfoService;
+import mcgillar.backend.services.user.AppUserService;
+
+@AllArgsConstructor
+@RestController
+@RequestMapping("/user-info/")
+public class AppUserInfoController {
+    
+    private AppUserInfoService appUserInfoService;
+    private AppUserService appUserService;
+
+    /**
+     * 
+     * @param firstName
+     * @param lastName
+     * @param authentication
+     * @return Updated User Info
+     * 
+     * If the parameter is not null or empty it will be set
+     */
+    @PostMapping("first-last-name")
+    public ResponseEntity<UserInfoTO> addOrModifyFirstNameOrLastName(
```

Comment:
Ik its a bit late now, but couldnt we just have one controller that updates all the info attributes by intaking a userinfo object from the frontend. That way we only need one update button on our profile page instead of 3 or needing to have the frontend send multiple requests

Reply:
Matthew requested it like this so i think its fine.

Reply:
Sounds good